### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Add `EventTransformation`, `BrokerTrigger` and `BrokerWithManyTriggers` e2e tests
- Fix related bugs

### CODE REVIEW

**Comment by pierDipi** [comment]
`event_info_store.go:105: Failed to start the recordevent pod 'recordevents-pod': timed out waiting for the condition`

/retest

**Comment by slinkydeveloper** [comment]
@pierDipi did you added the build test images script and configured to run in install phase?

**Comment by pierDipi** [comment]
Is [this](https://github.com/knative-sandbox/eventing-kafka-broker/pull/48/files#diff-69ef2e794e83216f7e132e4d4330e347R24-R33) enough?

**Comment by pierDipi** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project generated: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test failed: Plugin org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-surefire-plugin:jar:3.0.0-M5 -> org.apache.maven.surefire:maven-surefire-common:jar:3.0.0-M5: Failed to read artifact descriptor for org.apache.maven.surefire:maven-surefire-common:jar:3.0.0-M5: Could not transfer artifact org.apache.maven.surefire:maven-surefire-common:pom:3.0.0-M5 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/surefire/maven-surefire-common/3.0.0-M5/maven-surefire-common-3.0.0-M5.pom 502 Gateway Error -> [Help 1]
```

/retest

**Comment by pierDipi** [comment]
Ok, this is ready now.

/assign slinkydeveloper

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/EventMatcher.java [hunk 1]
```diff
@@ -101,6 +107,10 @@ public boolean match(final CloudEvent event) {
     return stringProvider.apply(s);
   }
 
+  private static boolean isNotAny(final String value) {
```
Comment (slinkydeveloper) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/EventMatcher.java:110]: maybe `notEmpty`?

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/EventMatcher.java:110]: Yeah, I was using the same terminology of the docs, but not a strong opinion here.

Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/EventMatcher.java]: Done: https://github.com/knative-sandbox/eventing-kafka-broker/pull/48/commits/54a3b82bf87698bc7c3b676640a64d97789bbd67

**Pull request merged**